src/main/scala/org/apache/spark/sql/delta/schema/ImplicitMetadataOperation.scala

```diff
@@ -79,8 +79,7 @@ trait ImplicitMetadataOperation extends DeltaLogging {
     def isNewSchema: Boolean = txn.metadata.schema != mergedSchema
     // We need to make sure that the partitioning order and naming is consistent
     // if provided. Otherwise we follow existing partitioning
-    def isNewPartitioning: Boolean = normalizedPartitionCols.nonEmpty &&
-      txn.metadata.partitionColumns != normalizedPartitionCols
+    def isNewPartitioning: Boolean = txn.metadata.partitionColumns != normalizedPartitionCols
```

Comment:
you need a more intricate condition here:
 1. `normalizedPartitionCols.nonEmpty && txn.metadata.partitionColumns != normalizedPartitionCols` OR
 2. `overwriteSchema &&  txn.metadata.partitionColumns != normalizedPartitionCols`

Reply:
i indroduced new variable for it